internal/kafka/internal/services/kafka.go

```diff
@@ -728,15 +734,20 @@ func (k *kafkaService) VerifyAndUpdateKafkaAdmin(ctx context.Context, kafkaReque
 		return errors.New(errors.ErrorValidation, fmt.Sprintf("Unable to update kafka: %s ibp version: %s with kafka version: %s", kafkaRequest.ID, kafkaRequest.DesiredKafkaIBPVersion, kafkaRequest.DesiredKafkaVersion))
 	}
 
-	vCompKafka, ek := api.CompareSemanticVersionsMajorAndMinor(kafkaRequest.ActualKafkaVersion, kafkaRequest.DesiredKafkaVersion)
+	currentKafkaVersion := kafkaRequest.ActualKafkaVersion
+	if currentKafkaVersion == "" { // the kafka version could be empty is the Kafka that have not yet been successful provisioned at first place
+		currentKafkaVersion = kafkaRequest.DesiredKafkaVersion // use the desired version for checks as eventually the actual version will be this one for new Kafkas
+	}
```

Comment:
We have code like this often in the code.
WDYT about creating an utility function like this:
```go
func OneOf(values ...string) (string, error) {
	for _, val := range values {
		if val != "" {
			return val, nil
		}
	}
	return "", fmt.Errorf("No value found")
}
```

so that previous code would become:
```go
currentKafkaVersion, _ := utils.OneOf(kafkaRequest.ActualKafkaVersion, kafkaRequest.DesiredKafkaVersion)
```

Reply:
That sounds good, reminds me of [lodash#get](https://lodash.com/docs/#get)

It is something I won't consider adding this to this PR but it's worth a discussion in the technical discussion call and see if it is something that can be widely adopted and added as a pattern. WDYT?

Reply:
@machi1990 agree